### PR TITLE
Add the garnix cache to the nix configuration if nix is already installed

### LINKED ISSUE #272: Installation script: Set up nix cache when nix is already installed too
The script only adds the cache when nix isn’t already installed. Likely it should ask for permission then add it itself.

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes https://github.com/garnix-io/garn/issues/272.

### CODE REVIEW

File: install-template.sh [hunk 1]
```diff
@@ -24,6 +24,34 @@ install_nix () {
...
+  # This ignores errors, for single-user installations.
+  if test "$(uname)" = "Linux" ; then
+    sudo systemctl restart nix-daemon.service || true
+  elif test "$(uname)" = "Darwin" ; then
+    sudo launchctl kickstart -k system/org.nixos.nix-daemon || true
+  else
+    echo "Unknown system: $(uname)"
+    echo Cancelling installation.
+    exit 1
+  fi
```
Comment (soenkehahn) [inline install-template.sh:48]: I tried to detect whether a nix install is a multi user or single user installation. `grep nixbld /etc/group` works, but only on linux. I pushed [this branch](https://github.com/garnix-io/nix-tool-installer/compare/sh/cache...sh/check-for-multi-user?expand=1) that does attempt to do a portable check. But it's really a lot of complexity given that it doesn't have automated tests and is written in bash. So whether that code really works is anyone's guess. So I would merge the version here. :shrug:

File: install-template.sh [hunk 2]
```diff
@@ -40,13 +68,31 @@ else
...
+echo BINARY CACHE CONFIGURATION
+echo ==========================
+
+if test_cache_configured; then
+  echo Hooray, the garnix cache is configured.
+else
+  echo For \'@@toolName@@\' to work well, it is recommended that you add the
+  echo garnix.io binary cache to your nix configuration. But the cache
+  echo configuration cannot be found.
+  printf "Should I add the garnix.io binary cache to /etc/nix/nix.conf now (requires sudo)? [y/n] "
```
Comment (jkarni) [inline install-template.sh:87]: I think we should say somehow that we’ll otherwise abort the installation.

File: install-template.sh (hunk 1 above)
Comment (jkarni) [reply install-template.sh:48]: I think “|| true” is good enough!

File: install-template.sh (hunk 2 above)
Comment (jkarni) [inline install-template.sh:87]: Why sometimes echo sometimes printf?

File: install-template.sh [hunk 3]
```diff
@@ -24,6 +24,34 @@ install_nix () {
   set -eu
 }
 
+test_cache_configured () {
+  nixConfig=$(nix --extra-experimental-features 'nix-command flakes' show-config) &&
+  (echo "$nixConfig" | grep --quiet "^substituters.*cache\.garnix\.io") &&
```
Comment (jkarni) [inline install-template.sh:29]: This won’t match “extra-substituers”, just “substitutes”, no? Shouldn’t the ^ be removed?

**Review by jkarni — APPROVED** [review]
Besides the ^ issue with substituters and trusted public keys, LGTM

File: install-template.sh (hunk 2 above)
Comment (soenkehahn) [reply install-template.sh:87]: `printf` doesn't print a trailing newline. So for questions it's `printf`. (I first used `echo -n`, but `shellcheck` says that that's not portable.)

File: install-template.sh (hunk 3 above)
Comment (soenkehahn) [reply install-template.sh:29]: No, `nix show-config` doesn't show the `extra-{substituters,trusted-public-keys}` fields, but only the aggregated results. So in there it actually says `^substituters: <list of all substituters, regardless of how they're added>`.

File: install-template.sh (hunk 2 above)
Comment (soenkehahn) [reply install-template.sh:87]: I tweaked the whole section. It doesn't *explicitly say that it'll abort if you say no, but it implies it more heavily. Also it mirrors the prompt for the nix question now. In 573881fd. Is that good?

Comment (jkarni) [reply install-template.sh:87]: Yup

**Pull request merged**